web/src/main/ui/app/components/flows-new/edit-flow/ui/new-step-dialog-ui.component.ts

```diff
@@ -28,25 +28,46 @@ export class NewStepDialogUiComponent implements OnInit {
   public stepType: typeof StepType = StepType;
   readonly stepOptions = Object.keys(this.stepType);
 
-  selectedSource = '';
   newStepForm: FormGroup;
   databases: any = [];
+  type: string = null;
+  isIngestion: boolean = false;
+  isMapping: boolean = false;
+  isMastering: boolean = false;
+  isCustom: boolean = false;
+  sourceRequired: boolean = false;
+  entityRequired: boolean = false;
+  hasSelectedCollection: boolean = false;
+  hasSelectedQuery: boolean = false;
+
   constructor(
     private formBuilder: FormBuilder) {}
 
   ngOnInit() {
+    let selectedSource;
     this.databases = Object.values(this.databaseObject).slice(0, -1);
     if (this.step) {
       this.newStep = this.step;
     }
+    if (this.step && this.step.selectedSource)
+      selectedSource = this.step.selectedSource;
+    else {
+      if (this.step && this.step.options && this.step.options.sourceQuery && !this.step.options.sourceCollection)
+        selectedSource = 'query';
+      else if (this.step && this.step.options && this.step.options.sourceCollection)
+        selectedSource = 'collection';
+    }
+    this.hasSelectedCollection = selectedSource === 'collection';
+    this.hasSelectedQuery = selectedSource === 'query';
     this.newStepForm = this.formBuilder.group({
       name: [this.step ? this.step.name : '', [
         Validators.required,
         Validators.pattern('[a-zA-Z][a-zA-Z0-9\_\-]*'),
         ExistingStepNameValidator.forbiddenName(this.flow, this.step && this.step.name)
       ]],
       stepDefinitionType: [this.step ? this.step.stepDefinitionType : '', Validators.required],
-      description: [(this.step && this.step.options.description) ? this.step.description : ''],
+      description: [(this.step && this.step.options.description) ? this.step.options.description : ''],
+      selectedSource: [(selectedSource) ? selectedSource : ''],
```

Comment:
I don't think description is inside the options object.

Reply:
Ok, thanks.  One or the other was wrong, I picked the wrong one.  ;)